ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -378,14 +376,14 @@ static int ompi_osc_rdma_component_query (struct ompi_win_t *win, void **base, s
         return -1;
     }
 
-#if OPAL_CUDA_SUPPORT
     /* GPU buffers are not supported by the rdma component */
     if (MPI_WIN_FLAVOR_CREATE == flavor) {
-        if (opal_cuda_check_bufs(*base, NULL)) {
+        int dev_id;
+        uint64_t flags;
+        if (opal_accelerator.check_addr(*base, &dev_id, &flags)) {
```

Comment:
I think the real question for this component (and some other components in ompi) is not necessarily 'is it a device buffer', but 'can we execute operations x/y/z on this address?' This could be achieved using the flags for the check_addr() function by defining the corresponding flags in accelerator.h, or we could introduce a different interface (aka check_capabilities() or similar.)

Reply:
This was sorted out offline through slack